### PR TITLE
Update dependencies and move eslint to dev dependencies

### PR DESCRIPTION
**Pull request by lemonmade**
For some reason had ESLint as a dependency here :S

### CODE REVIEW

File: icon-loader.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ module.exports = function shopifyIconLoader(source) {
     return fill.includes('#FFF') ? 'fill="currentColor"' : '';
   });
 
+  // Issue with ESLint recognizing this as needing an object destructure
+  // eslint-disable-next-line prefer-destructuring
```
Comment (lemonmade) [inline icon-loader.js]: @ismail-syed any idea what's going on here?

Comment (ismail-syed) [inline icon-loader.js]: It's related to the [`enforceForRenamedProperties`](https://github.com/Shopify/eslint-plugin-shopify/blob/master/lib/config/rules/ecmascript-6.js#L43) we have enabled. 

I think what's happening is the rule is expecting something like
```
const [, viewBox] = VIEWBOX_REGEX.exec(finalSource);
```

See [docs](http://eslint.org/docs/rules/prefer-destructuring#options)

Comment (lemonmade) [inline icon-loader.js]: But we have array restructuring set to be false, no?

Comment (ismail-syed) [inline icon-loader.js]: It seems like the [`enforceForRenamedProperties`](https://github.com/eslint/eslint/blob/master/lib/rules/prefer-destructuring.js#L117-L119) option doesn't take into consideration the state of the array option.

Comment (ismail-syed) [inline icon-loader.js]: @lemonmade 
https://github.com/eslint/eslint/issues/8654#issuecomment-304184244

**Pull request merged**